src/main/java/com/mathworks/ci/RunMatlabCommandBuilder.java

```diff
@@ -112,22 +116,45 @@ public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
     private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
             TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
         final String uniqueTmpFldrName = getUniqueNameForRunnerFile();
-        ProcStarter matlabLauncher;
+        final String uniqueCommandFile =
+                "command_" + getUniqueNameForRunnerFile().replaceAll("-", "_");
+        final FilePath uniqeTmpFolderPath =
+                getFilePathForUniqueFolder(launcher, uniqueTmpFldrName, workspace);
+        
+        // Create MATLAB script
+        createMatlabScriptByName(uniqeTmpFolderPath,uniqueCommandFile,workspace,listener);
+
         try {
-            matlabLauncher = getProcessToRunMatlabCommand(workspace, launcher, listener, envVars,
-                    getCommand(), uniqueTmpFldrName);
-            return matlabLauncher.join();
+            // Start the launcher from temp folder
+            ProcStarter matlabLauncher = launcher.launch().pwd(uniqeTmpFolderPath).envs(envVars);
+            listener.getLogger()
+                    .println("#################### Starting command output ####################");
+            return getProcessToRunMatlabCommand(matlabLauncher, workspace, launcher, listener,
+                    uniqueCommandFile, uniqueTmpFldrName).join();
+
         } catch (Exception e) {
             listener.getLogger().println(e.getMessage());
             return 1;
         } finally {
-            // Cleanup the runner File from tmp directory
-            FilePath matlabRunnerScript =
-                    getFilePathForUniqueFolder(launcher, uniqueTmpFldrName, workspace);
-            if (matlabRunnerScript.exists()) {
-                matlabRunnerScript.deleteRecursive();
+            // Cleanup the tmp directory
+            if (uniqeTmpFolderPath.exists()) {
+                uniqeTmpFolderPath.deleteRecursive();
             }
         }
-
+    }
+    
+    private void createMatlabScriptByName(FilePath uniqeTmpFolderPath, String uniqueScriptName, FilePath workspace, TaskListener listener) throws IOException, InterruptedException {
+     
+        // Create a new command runner script in the temp folder.
+        final FilePath matlabCommandFile =
+                new FilePath(uniqeTmpFolderPath, uniqueScriptName + ".m");
+        final String matlabCommandFileContent =
+                "cd '" + workspace.getRemote().replaceAll("'", "''") + "';\n" + getCommand();
+
+        // Display the commands on console output for users reference
+        listener.getLogger()
+                .println("Generating MATLAB script with content:\n" + getCommand() + "\n");
+
+        matlabCommandFile.write(matlabCommandFileContent, "UTF-8");

```

Comment:
There is still the open question of UTF-8 support for MATLAB in older releases. I am okay if we just document the limitation.

Reply:
I will discuss with @mw-hrastega and make sure if its get documented at the least

Reply:
Will a UTF-8 encoded file with ascii characters still work in older releases? Or are we just talking abut the case where someone includes a non-ascii character in the command? If its the latter then I  am fine with the limitation. If its the former that is a blocking issue. If we are writing ascii characters to the file as UTF-8 will it work in older releases? If not we should encode it differently

Reply:
My understanding is Unicode if fully backward-compatible with ASCII (ASCII is a subset of Unicode). So UTF8 files should work fine in older releases as long as the script contains only ASCII characters.

If a user enters a command with a non-ASCII character, it will not work correctly in older releases. But it will not work correctly in older releases even if we write the file as ASCII.

Reply:
I am good writing it as unicode as long as it is qualified against at least one release where MATLAB didn't support it for m-files.

Reply:
it will be a great feature for new releases to have a Japanese hello world for example

Reply:
I demonstrated the implications to Nikhil today. First of all, if the users are using ASCII text only then all this is moot. All our supported encodings use the same binary representation for ASCII. 

The problem occurs if the users use non-ASCII characters. For instance a MATLAB user in Japan who uses Japanese characters in their command. e.g. 'disp("こんにちは"). If the user in Japan is using a Japanese locale Windows machine (a very likely case), then MATLAB prior to R2020a will expect the .m files to be encoded the SHIFT-JIS encoding. Executing the UTF-8 encoded file will produce character corruption. In MATLAB 2020a and newer, the utf-8 encoded file with Japanese characters will be executed correctly.
However, this is not a trivial problem to solve for 19b and older, as we will have to ask MATLAB what the right .m encoding (Shift-JIS on a Japanese Windows,  ISO-8859-1 on an English Windows and so on) will be. For this we will have to actually run MATLAB and execute some commands. :-)

Reply:
Updated the help doc for custom MATLAB command 

![image](https://user-images.githubusercontent.com/47204011/85831517-0c7d0800-b7ac-11ea-83b4-30ccaa3394ea.png)